src/include/cursor.i

```diff
@@ -190,6 +190,7 @@ __cursor_reset(WT_CURSOR_BTREE *cbt)
 		if (!F_ISSET(cbt, WT_CBT_NO_TXN))
 			__cursor_leave(session);
 		F_CLR(cbt, WT_CBT_ACTIVE);
+		F_CLR(cbt, WT_CBT_RETRY_NEXT | WT_CBT_RETRY_PREV);
```

Comment:
I think a simpler/cleaner approach to this would be to add `WT_CBT_RETRY_NEXT | WT_CBT_RETRY_PREV` to `WT_CBT_POSITION_MASK`.